Fix "Failed: undefined" Flake

U8NWXD opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #[n/a].
2. This PR does the following: Fixes the `Failed: undefined` flake.

## Essential Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct

Reruns of the `e2e_translation_classroom_and_core_features` workflow:

1. Pass: https://github.com/oppia/oppia/runs/7771758310?check_suite_focus=true
2. Pass: https://github.com/oppia/oppia/runs/7773865401?check_suite_focus=true
3. Pass: https://github.com/oppia/oppia/runs/7774898554?check_suite_focus=true
4. Failed with `option:checked` flake: https://github.com/oppia/oppia/runs/7776674851?check_suite_focus=true
5. Failed due to lost connection with GitHub runner: https://github.com/oppia/oppia/runs/7777922125?check_suite_focus=true
6. Pass: https://github.com/oppia/oppia/runs/7780254184?check_suite_focus=true
7. Failed with `Failed: First RTE Element` failure, which is the updated message for `Failed: undefined`: https://github.com/oppia/oppia/runs/7789795144?check_suite_focus=true

Reruns after fixing the failure in run 7 above with feaa707300635b6c13ec790f9735cbeba8099583:

1. Pass: https://github.com/oppia/oppia/runs/7800376623?check_suite_focus=true
2. Pass: https://github.com/oppia/oppia/runs/7808652818?check_suite_focus=true
   * After this run, pushed 94ea25883ef72154fac0ef9cf0920b620506e1d5 to set `expectFadeIn=true` in voiceover tests, which are unrelated to the `e2e_translation_classroom_and_core_features` workflow being tested here
3. Pass: https://github.com/oppia/oppia/runs/7810248101?check_suite_focus=true
4. Pass: https://github.com/oppia/oppia/runs/7812647388?check_suite_focus=true
   * After this run, pushed 1db1ccbe3a574c897cfc3c91235d131479463ebf to set `expectFadeIn=true` in more places
5. Pass: https://github.com/oppia/oppia/runs/7814185028?check_suite_focus=true
6. Pass: https://github.com/oppia/oppia/runs/7815685495?check_suite_focus=true
7. Pass: https://github.com/oppia/oppia/runs/7816178048?check_suite_focus=true
8. Pass: https://github.com/oppia/oppia/runs/7816894819?check_suite_focus=true
9. Pass: https://github.com/oppia/oppia/runs/7817541830?check_suite_focus=true
10. Failed with `Hosted Agent lost communication with the server.`: https://github.com/oppia/oppia/runs/7820480435?check_suite_focus=true
11. Pass: https://github.com/oppia/oppia/actions/runs/2849551459

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).
- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL". Oppiabot will help assign that person for you.
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).

U8NWXD committed:
Replace waitFor.fadeInToComplete w/ browser.sleep

Attempted to get fadeInToComplete to work in
https://github.com/U8NWXD/oppia/pull/18 but failed, so reverted to using
browser.sleep() instead.

Fixes the `Failed: undefined` flake.

U8NWXD committed:
Add messages for waitFor calls in forms.js

ShivamJhaa commented:
@U8NWXD The replacement for `browser.sleep()` in webdriverio is `browser.pause()`. Please update this in the webdriverio files. This will fix the following [error](https://github.com/oppia/oppia/runs/7759420929?check_suite_focus=true) in webdriverio suites.
Thanks.

U8NWXD committed:
Use browser.pause for webdriverIO

SAEb-ai commented on core/tests/protractor_utils/forms.js:
```diff
@@ -194,7 +194,10 @@ var RichTextEditor = async function(elem) {
     '.e2e-test-close-rich-text-component-editor');
   // Set focus in the RTE.
-  await waitFor.elementToBeClickable(rteElements.first());
+  await waitFor.elementToBeClickable(
+    rteElements.first(),
+    'First RTE element'
+  );
```
This part of the code is giving [this ](https://github.com/oppia/oppia/runs/7777125375?check_suite_focus=true#step:13:579) error now. PTAL. Also, the error message doesn't seen to be descriptive. It should be something like `First RTE element is not clickable`.

Ditto in the webdriverio code.

U8NWXD replied:
done

U8NWXD committed:
Set expectFadeIn to true for more setContent calls

expectFadeIn needs to be true whenever
ExplorationEditorMainTab.setContent is called after
ExplorationEditorMainTab.setStateName.

U8NWXD committed:
Make waitFor error message clearer

U8NWXD committed:
Set expectFadeIn=true in voiceoverUploadFeatures

U8NWXD committed:
Set expectFadeIn=true Everywhere Needed

Wherever we call ExplorationEditorMainTab.setContent immediately after
calling ExplorationEditorMainTab.setStateName, set `expectFadeIn=true`.

SAEb-ai commented on core/tests/protractor_utils/forms.js:
```diff
@@ -194,7 +194,10 @@ var RichTextEditor = async function(elem) {
     '.e2e-test-close-rich-text-component-editor');
   // Set focus in the RTE.
-  await waitFor.elementToBeClickable(rteElements.first());
+  await waitFor.elementToBeClickable(
+    rteElements.first(),
+    'First RTE element taking too long to become clickable.'
+  );
   await (await rteElements.first()).click();
```
Why are we not using `action.click` to peform this?

U8NWXD replied:
Just because we haven't migrated `forms.js` to use `action` functions yet

SAEb-ai reviewed (approved):
Approving based on the severity of the flake. But left one comment too. PTAL.
